### PR TITLE
Refactor: Proper callbacks, better error and dsl

### PR DESCRIPTION
**Pull request by getand**
- [x] Add proper callbacks
- [x] Add logger functionality 
- [x] Ensure error has all options (request-headers, x-request-id, ..)
- [x] Update Readme
- [x] Inherit logger from ancestor chain

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -69,7 +70,7 @@ GEM
     zeitwerk (2.4.2)
 
 PLATFORMS
-  ruby
+  x86_64-darwin-19
```
Comment (nsommer) [inline Gemfile.lock:73]: Intentional change?

File: lib/nxt_http_client/error.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ def initialize(response, message = nil)
     alias_method :to_s, :message
 
     def default_message
-      "NxtHttpClient::Error::#{response_code}"
+      "#{self.class.name}::#{response_code}"
```
Comment (nsommer) [inline lib/nxt_http_client/error.rb:16]: This will improve sentry readability 👍

**Review by nsommer — APPROVED** [review]
🎉

File: Gemfile.lock (hunk 1 above)
Comment (getand) [reply Gemfile.lock:73]: Probably comes from a new bundler version?

**Pull request merged**